.NET/Microsoft.Recognizers.Text.DateTime/Parsers/BaseDateTimeParser.cs

```diff
@@ -251,6 +251,21 @@ private DateTimeResolutionResult MergeDateAndTime(string text, DateObject refere
 
             ret.FutureValue = DateObject.MinValue.SafeCreateFromValue(futureDate.Year, futureDate.Month, futureDate.Day, hour, min, sec);
             ret.PastValue = DateObject.MinValue.SafeCreateFromValue(pastDate.Year, pastDate.Month, pastDate.Day, hour, min, sec);
+
+            // Handle case like "Wed Oct 26 15:50:06 2016" which year and month separated by time.
+            var matchYear = this.config.YearRegex.Match(text.Substring(er2[0].Start + er2[0].Length ?? 0));
+            if (matchYear.Success && pr1.TimexStr.Substring(0, 4) == "XXXX" && pr1.TimexStr.Substring(5, 2) != "XX")
```

Comment:
Can you add a code comment here explaining what this condition means?
Also, don't add hardcoded timex terms. Declare them in some of the existing Timex classes and use here. This makes it easier to maintain in the future, especially if timex expressions change.
The substring comparisons should also be methods in one of the existing timex classes (and substring is not the best way to do this. If XXXX changes to something else, it will break this check, for example.